EIPS/eip-4626.md

```diff
@@ -114,16 +114,17 @@ MUST _NOT_ revert.
 
 #### previewDeposit
 
-Allows an on-chain or off-chain user to simulate the effects of their deposit at the current block,
-given current on-chain conditions.
+Allows an on-chain or off-chain user to simulate the effects of their deposit at the current block, given current on-chain conditions.
+
+MUST return as close to and no more than the exact amount of Vault shares that would be minted in a `deposit` call in the same transaction. I.e. `deposit` should return the same or more `shares` as `previewDeposit` if called in the same transaction.
+
+MUST NOT account for deposit limits like those returned from maxDeposit and should always act as though the deposit would be accepted, regardless if the user has enough tokens approved, etc.
 
-MUST return the *exact* amount of Vault shares that would be minted if the caller were to deposit
-a given *exact* amount of underlying assets using the `deposit` method.
 MUST be inclusive of deposit fees. Integrators should be aware of the existence of deposit fees.
-MUST _NOT_ revert.
 
-Note that any discrepancy between `shares * assetsPerShare` and `assets` SHOULD be considered slippage
-in share price or some other type of condition, meaning the depositor will lose assets by depositing.
+MUST NOT revert.
+
+Note that any unfavorable discrepancy between `convertToShares` and `previewDeposit` SHOULD be considered slippage in share price or some other type of condition, meaning the depositor will lose assets by depositing.
```

Comment:
Note that `convertToAssets` doesn't exist yet

Reply:
It does now!